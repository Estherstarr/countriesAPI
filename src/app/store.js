import { configureStore } from '@reduxjs/toolkit';
import fetchAllReducerSlice from '../redux/AllCountries';

const store = configureStore({
  reducer: {
    allCountries: fetchAllReducerSlice,
  },
});

export default store;
