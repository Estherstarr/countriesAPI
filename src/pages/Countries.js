import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchAllCountries } from '../redux/AllCountries';
import Header from '../components/Header';
import SearchBar from '../components/SearchBar';
import CountryList from '../components/CountryList';

export default function Countries() {
  const allState = useSelector((state) => state.allCountries);
  const dispatch = useDispatch();
  const [query, setQuery] = useState('');

  useEffect(() => {
    dispatch(fetchAllCountries());
  }, []);

  const handleChange = (e) => {
    setQuery(e.target.value);
  };
  const filteredCountries = allState.data.filter((country) => country.name.common.includes(query));
  return (
    <div className="h-full">
      <Header />
      <div className="p-10 md:p-20">
        <SearchBar onChange={handleChange} />
        <div className="grid grid-cols-1 md:grid-cols-4 gap-16 w-full">
          {!allState.pending && <CountryList countries={filteredCountries} />}
        </div>
      </div>
    </div>
  );
}
