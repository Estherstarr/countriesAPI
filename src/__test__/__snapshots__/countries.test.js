import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../app/store';
import Countries from '../../pages/Countries';

describe('display countries components', () => {
  test('test if countries component renders in the dom', () => {
    render(
      <Provider store={store}>
        <Countries />
      </Provider>,
    );
    expect(screen.queryByText(/Where is the world?/)).toBeTruthy();
  });
  test('snapshot renders properly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Countries />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
