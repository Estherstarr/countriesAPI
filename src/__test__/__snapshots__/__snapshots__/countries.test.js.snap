// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display countries components snapshot renders properly 1`] = `
<div
  className="h-full"
>
  <div
    className="bg-[#2d4474] shadow-lg py-3 px-10 flex justify-between"
  >
    <h1
      className="font-bold text-white"
    >
      Where is the world?
    </h1>
    <h3
      className="text-white"
    >
      Dark Mode
    </h3>
  </div>
  <div
    className="p-10 md:p-20"
  >
    <div
      className="mb-10 flex flex-col gap-y-3 md:flex-row justify-between "
    >
      <input
        className="w-160 md:w-80 rounded p-2 shadow-lg "
        onChange={[Function]}
        placeholder="Search for a country"
        type="text"
      />
    </div>
    <div
      className="grid grid-cols-1 md:grid-cols-4 gap-16 w-full"
    />
  </div>
</div>
`;
